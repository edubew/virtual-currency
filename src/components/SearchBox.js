import React from 'react';
import { FiSearch } from 'react-icons/fi';
import PropTypes from 'prop-types';

const SearchBox = ({ search, setSearch }) => (
  <form>
    <FiSearch />
    <input
      className="search__input"
      type="text"
      value={search}
      onChange={(e) => setSearch(e.target.value.toLowerCase())}
      placeholder="Search Crypto by name..."
      maxLength={50}
    />
  </form>
);

SearchBox.propTypes = {
  search: PropTypes.string.isRequired,
  setSearch: PropTypes.string.isRequired,
};

export default SearchBox;
