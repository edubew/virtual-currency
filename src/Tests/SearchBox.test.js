import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import SearchBox from '../components/SearchBox';

describe('Test SearchBox Component', () => {
  it('should render SearchBox component', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <SearchBox />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
