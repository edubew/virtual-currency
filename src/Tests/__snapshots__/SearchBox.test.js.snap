// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchBox Component should render SearchBox component 1`] = `
<form
  className="sc-bcXHqe jEOLbp"
>
  <svg
    fill="none"
    height="1em"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="11"
      cy="11"
      r="8"
    />
    <line
      x1="21"
      x2="16.65"
      y1="21"
      y2="16.65"
    />
  </svg>
  <input
    className="search__input"
    maxLength={50}
    onChange={[Function]}
    placeholder="Search by name..."
    type="text"
  />
</form>
`;
